doc/operator-backup-and-restore.md

```diff
@@ -0,0 +1,130 @@
+# 3scale installation Backup and Restore using the operator
+
+## Table of contents
+* [General description](#general-description)
+* [Backing up 3scale](#backing-up-3scale)
+  * [Backup compatible scenarios](#restore-compatible-scenarios)
+  * [Backup workflow](#backup-workflow)
+* [Restoring 3scale](#restoring-3scale)
+  * [Restore compatible scenarios](#restore-compatible-scenarios)
+  * [Restore workflow](#restore-workflow)
+* [APIManagerBackup CRD reference](apimanagerbackup-reference.md)
+* [APIManagerRestore CRD reference](apimanagerrestore-reference.md)
+
+## General description
+
+Backup and Restore functionality of a 3scale installation is provided for
+3scale installations deployed using the [APIManager](apimanager-reference.md)
+custom resource definition provided by the 3scale-operator.
+
+Operator capabilities custom resources are not part of the 3scale Installation
+functionality and are thus not included as part of the 3scale installation
+Backup and Restore functionality.
+
+To see how to back up an APIManager 3scale based installation see [#Backing up 3scale](#backing-up-3scale)
+
+To see how to restore a previously backed up APIManager 3scale based installation
+using the operator backup functionality see [restoring 3scale](#restoring-3scale)
+
+## Backing up 3scale
+
+The backup functionality of a 3scale installation deployed by an `APIManager`
+custom resource is provided.
+
+### Backup compatible scenarios
+
+To see what 3scale installation configurations can be backed up see the
+[APIManagerBackup reference](apimanagerbackup-reference.md). Specifically the following sections should be
+consulted:
+* [Backup scenarios scope](apimanagerbackup-reference.md#backup-scenarios-scope)
+* [Data that is backed up](apimanagerbackup-reference.md#data-that-is-backed-up)
+
+### Backup workflow
+
+To backup a 3scale installation deployed with an existing APIManager the
+workflow is the following one:
+
+1. Perform a backup of the 3scale external databases:
+   * backend-redis
+   * system-redis
+   * system database (MySQL or PostgreSQL)
+1. Perform a backup of the following Kubernetes secrets:
+   * backend-redis
+   * system-redis
+   * system-database
+1. Create the APIManagerBackup Custom resource in the same namespace
+   as where the 3scale installation managed by the APIManager object
+   is deployed. See the [APIManagerBackup reference](apimanagerbackup-reference.md)
+   to see the available fields that can be configured. An example would be:
+   ```
+     apiVersion: apps.3scale.net/v1alpha1
+     kind: APIManagerBackup
+     metadata:
+      name: example-apimanagerbackup-pvc
+     spec:
+       backupDestination:
+         persistentVolumeClaim:
+           resources:
+             requests: "10Gi"
+   ```
+1. Wait until APIManagerBackup finishes. You can check this by obtaining
```

Comment:
`oc wait --for=condition=available --timeout=-1s APIManagerBackup/example-apimanagerbackup-pvc`

Reply:
This is not a DeploymentConfig.
There's no "available" condition / conditions.

Reply:
backup type has condition list. Is not used at all? then shoud be removed?

Reply:
Removed